the-collab-lab/tcl-61-smart-shopping-list
# Styling

## Description

- Implement Sass styling to the application to match wireframe (light & dark mode)
- Installed `react-svg `to be able to import `svg` as a react component
- Improve code consistency throughout the application (e.g., the addition of `useTimeout` for all success/error messages)
- Relocated success & error messages to render below the input 
- Reorganized the ListItem component to display all relevant information (item, item urgency, and delete button) on a single line
- Ensure application responsiveness by implementing media queries


## Related Issue

Closes #13 

## Acceptance Criteria

- A consistent, dazzling user experience.

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|  ✓   | :hammer: Refactoring       |
|  ✓   | 🎨 : Styling       |
|     | :100: Add tests            |
|   ✓  | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before

<img width="536" alt="Screenshot 2023-05-26 at 9 41 48 AM" src="https://github.com/the-collab-lab/tcl-61-smart-shopping-list/assets/29443697/d5ac1d9a-2139-4f59-8975-bf229079b6b5">

### After

https://github.com/the-collab-lab/tcl-61-smart-shopping-list/assets/97640502/1e73b8d0-0cb8-4bba-b392-a0fcd549ecc8


https://github.com/the-collab-lab/tcl-61-smart-shopping-list/assets/97640502/d22baaf5-dace-427d-a747-9f22b60f6097


**Include gif demo here**

## Testing Steps / QA Criteria
1. Ensure that you have executed `npm install` 
2. Run the app and have a look at the styling changes 👀 (can access dark mode by toggling system settings) 
3. Test the user journey and view all styling with the following steps:
- If you have an existing shopping list, clear the token to start with a fresh list.
- Open the application.
- Create a new list.
- Add multiple new items, and switch between the urgency levels ("soon," "kind of soon," and "not soon").
- Attempt to add an item that already exists on the list and verify that an appropriate error message is displayed.
- Delete an item from the list.
